### DISCUSSION TITLE
Add Adaptive Splitting Decision Trees to River

### DISCUSSION THREAD

**Discussion by danielnowakassis** [online-ml/river#1609] [Show and tell]
Hi.

I've implemented my decision tree in River and run all the tests for adding a new estimator.

The published paper is here: https://dl.acm.org/doi/10.1145/3605098.3635899

We proposed an adaptive splitting mechanism, where a change detector monitors the error rate or data distribution purity (parameter) of the leaf node to determine the split point (and if merit > 0). It had some good results against SOTA decision trees, and I have also tested it in real datasets in River. 

Accuracy : 

![accuracy](https://github.com/user-attachments/assets/92a22fe7-dc46-4b8d-a6a7-876095856526)

I am now writing an ablation study of the method. I'm open to any questions. LAST would be error rate monitoring, LAST_D would be data distribution purity monitoring.

Can I do a PR?

**Comment by MaxHalford** [comment]
Summoning the resident tree hugger @smastelini 🌴🤗

**Comment by smastelini** [comment]
Hi @danielnowakassis , sounds good to me! The idea also sounds really clever :)
